.github/ci.yml

```diff
@@ -0,0 +1,85 @@
+name: CI
+
+env:
+  JDK_JAVA_OPTIONS: -XX:+PrintCommandLineFlags # JDK_JAVA_OPTIONS is _the_ env. variable to use for modern Java
+  JVM_OPTS: -XX:+PrintCommandLineFlags # for Java 8 only (sadly, it is not modern enough for JDK_JAVA_OPTIONS)
+
+on:
+  pull_request:
+  push:
+    branches: ['master']
+  release:
+    types:
+      - published
+
+jobs:
+  lint:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 30
+    steps:
+      - name: Checkout current branch
+        uses: actions/checkout@v2.4.0
+        with:
+          fetch-depth: 0
+      - name: Setup Scala and Java
+        uses: olafurpg/setup-scala@v10
+      - name: Cache scala dependencies
+        uses: coursier/cache-action@v6
+      - name: Lint code
+        run: sbt ++3.1.3 check
+
+  test:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 30
+    strategy:
+      fail-fast: false
+      matrix:
+        java: ['adopt@1.8', 'adopt@1.11']
```

Comment:
Ok, we should probably add Java 17 here.
Why have you chosen "adopt" over graal or openjdk?

Reply:
This was laziness as I had other builds which used these, let me check what images GitHub provides.

Reply:
I myself run graal on my machine so maybe it'd be good to have the CI run openjdk.
That should suffice, I'd say.

Reply:
I'm actually tempted to use corretto for CI, though temurin seems viable. https://github.com/actions/setup-java#supported-distributions . If by OpenJDK you mean Oracle, it is not available and I think it is avoided because of Oracle's tendancies to be agressive with its license enforcement.

Reply:
It should be noted all the distributions listed at the link I provided use the monicker "OpenJDK"

Reply:
ok, fine.
Then simply choose whatever you deem best.